Code/GraphMol/RWMol.cpp

```diff
@@ -527,6 +527,13 @@ unsigned int RWMol::addBond(unsigned int atomIdx1, unsigned int atomIdx2,
   b->setBeginAtomIdx(atomIdx1);
   b->setEndAtomIdx(atomIdx2);
 
+  // we're in a batch edit, and at least one of the bond ends is scheduled
```

Comment:
This appears to be the "Fix" for the leak, correct?

Reply:
Yes. This will mark these bonds for deletion in the batch edit, so the bonds will be deleted if the batch edit is committed, or will stay if the batch commit is rolled back.